src/v3/message.rs

```diff
@@ -0,0 +1,260 @@
+//! Provides types related to [`crate::v3::Message`].
+
+use serde::Serialize;
+
+/// The settings to use when sending the [`crate::v3::Message`].
+/// See the [api docs](https://www.twilio.com/docs/sendgrid/api-reference/mail-send/mail-send#request-body)
+/// for details.
+#[derive(Default, Serialize)]
+pub struct MailSettings {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    bypass_list_management: Option<BypassListManagement>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    bypass_spam_management: Option<BypassSpamManagement>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    bypass_bounce_management: Option<BypassBounceManagement>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    bypass_unsubscribe_management: Option<BypassUnsubscribeManagement>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    footer: Option<Footer>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sandbox_mode: Option<SandboxMode>,
+}
+
+/// Used for the bypass list management setting.
+#[derive(Default, Serialize)]
+pub struct BypassListManagement {
+    enable: bool,
+}
+
+/// Used for the bypass spam management setting.
+#[derive(Default, Serialize)]
+pub struct BypassSpamManagement {
+    enable: bool,
+}
+
+/// Used for the bypass bounce management setting.
+#[derive(Default, Serialize)]
+pub struct BypassBounceManagement {
+    enable: bool,
+}
+
+/// Used for the bypass unsubscribe management setting.
+#[derive(Default, Serialize)]
+pub struct BypassUnsubscribeManagement {
+    enable: bool,
+}
+
+/// Used to provide a footer for the [`crate::v3::Message`].
+#[derive(Default, Serialize)]
+pub struct Footer {
+    enable: bool,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    text: Option<String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    html: Option<String>,
+}
+
+/// Used for the sandbox mode setting.
+#[derive(Default, Serialize)]
+pub struct SandboxMode {
+    enable: bool,
+}
+
+impl MailSettings {
+    /// Create a new default [`MailSettings`] instance.
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Set the bypass list management setting.
+    pub fn set_bypass_list_management(
+        mut self,
+        bypass_list_management: BypassListManagement,
+    ) -> Self {
+        self.bypass_list_management = Some(bypass_list_management);
+        self
+    }
+
+    /// Set the bypass spam management setting.
+    pub fn set_bypass_spam_management(
```

Comment:
I was reading through some of the [documentation](https://www.twilio.com/docs/sendgrid/ui/sending-email/index-suppressions#bypass-filters-and-v3-mail-send) and saw that a few of the bypass functions cannot be used if `bypass_list_management` is enabled. Do you think it's worth trying to prevent that as part of the library?

Reply:
Interesting. Yeah, I think that's a good idea and should be possible. I'll update this.

Reply:
Done.